File: 643-maximum-average-subarray-i/maximum-average-subarray-i.go
func findMaxAverage(nums []int, k int) float64 {
    var sum float64

    for i := 0; i < k; i++ {
        sum += float64(nums[i])
    }

    maxSum := sum

    for i := k; i < len(nums); i++ {
        sum += float64(nums[i]) - float64(nums[i-k])
        
        if sum > maxSum {
            maxSum = sum
        }
    }

    return maxSum / float64(k)
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function (nums, k) {
    var maxRes = 0
    let count = 0

    for (let i = 0; i < nums.length; i++) {
        
        if (i < k) {
            count += nums[i]
            if (i == k - 1) {
                maxRes = count / k
                continue
            }
        } else {
            count = (count + nums[i] - nums[i - k])
            maxRes = Math.max(count / k, maxRes)
        }
    }

    return maxRes
};